internal/controllers/gateway/route_utils.go

```diff
@@ -82,6 +84,12 @@ const (
 	// https://github.com/kubernetes-sigs/gateway-api/pull/1516
 	// TODO: swap this out with upstream const when released.
 	RouteReasonNoMatchingParent gatewayv1beta1.RouteConditionReason = "NoMatchingParent"
+	// This reason is used with the "Accepted" condition when no hostnames in listeners
+	// could match hostname in the spec of route.
+	// NOTE: This is not in upstream and still under discussion:
+	// https://github.com/kubernetes-sigs/gateway-api/issues/1561
+	// may be changed if upstream decide to use a different reason for this case.
+	RouteReasonNoMatchingListenerHostname gatewayv1beta1.RouteConditionReason = "NoMatchingListenerHostname"
```

Comment:
Since we've discovered that [it's already available upstream](https://github.com/kubernetes-sigs/gateway-api/blob/v0.5.1/apis/v1beta1/shared_types.go#L216-L218), why not use it here?

Reply:
updated.